pkgs/by-name/fr/frigate/package.nix

```
@@ -88,7 +88,7 @@ python.pkgs.buildPythonApplication rec {
         --replace-fail "/usr/local/lib/vec0" "${lib.getLib sqlite-vec}/lib/vec0${stdenv.hostPlatform.extensions.sharedLibrary}"
 
     ''
-    + lib.optionalString (stdenv.hostPlatform == "x86_64-linux") ''
+    + lib.optionalString stdenv.hostPlatform.isLinux ''
```

This changes the intent. `clang-rocm` is unsupported on anything but `x86_64-linux`.

---

Does this match the intent:

```suggestion
    + lib.optionalString (lib.meta.availableOn stdenv.hostPlatform rocmPackages.clr) ''
```

?

---

Unfortunately the check does not work for transitive dependencies.

---

I would be reluctant to match on the rocm-clang, since other packages in the chain could also be broken. And it needs to be `isx86_64 && isLinux`, because it is also not compatible with i686-linux for some reason.

---

> I would be reluctant to match on rocm-clang, since other packages in the chain could also be unsupported/broken as well.

To clarify, `rocmPackages.clang` is the transitive input that only works on x86_64-linux?

Sadly, we can't use `availableOn hostPlatform rocmPackages.clang` because that package doesn't declare `meta.platforms`. `availableOn` would always return `true`.